src/components/TagPopover.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import A from './A'
+import Avatar from './Avatar'
+import cx from 'classnames'
+import { isEmpty, get } from 'lodash'
+import { hideTagPopover, fetchTagSummary, followTag } from '../actions/index'
+const { string, object, func, array, number, bool } = React.PropTypes
+
+const nounCount = (n, noun) => `${n} ${noun}${Number(n) !== 1 ? 's' : ''}`
+
+@connect(({ tagPopover: { slug, tagName, position, anchorWidth }, tagsByCommunity }) => {
+  let tag = get(tagsByCommunity, [slug, tagName])
+  return {
+    tagName,
+    slug,
+    position,
+    anchorWidth,
+    followed: get(tag, 'followed'),
+    description: get(tag, 'description'),
+    postCount: get(tag, 'post_count'),
+    followerCount: get(tag, 'follower_count'),
+    activeMembers: get(tag, 'active_members')
+  }
+})
+export default class TagPopover extends React.Component {
+
+  static propTypes = {
+    dispatch: func,
+    tagName: string,
+    slug: string,
+    description: string,
+    postCount: string,
+    followerCount: number,
+    activeMembers: array,
+    position: object,
+    anchorWidth: number,
+    followed: bool
+  }
+
+  hide () {
+    let { dispatch } = this.props
+    dispatch(hideTagPopover())
+  }
+
+  removeHideTimeout () {
+    if (this.hideTimeoutId) {
+      clearTimeout(this.hideTimeoutId)
+      this.hideTimeoutId = null
+    }
+  }
+
+  componentWillReceiveProps (nextProps) {
+    let { dispatch, slug, tagName } = this.props
+
+    // hidden
+    if (isEmpty(nextProps.tagName)) return
+
+    // loading
+    if (nextProps.tagName !== tagName || nextProps.slug !== slug) {
+      dispatch(fetchTagSummary(nextProps.tagName, nextProps.slug))
+    }
+
+    // content has loaded
+    if (nextProps.description) {
+      this.hideTimeoutId = setTimeout(() => this.hide(), 3000)
```

Comment:
after 3 seconds, yes

On Tue, Jun 14, 2016 at 3:45 PM, Lawrence Wang <notifications@github.com>
wrote:

> In src/components/TagPopover.js
> <https://github.com/Hylozoic/hylo-redux/pull/63#discussion_r67071801>:
>
> > +  }
> > +
> > +  componentWillReceiveProps (nextProps) {
> > +    let { dispatch, slug, tagName } = this.props
> > +
> > +    // hidden
> > +    if (isEmpty(nextProps.tagName)) return
> > +
> > +    // loading
> > +    if (nextProps.tagName !== tagName || nextProps.slug !== slug) {
> > +      dispatch(fetchTagSummary(nextProps.tagName, nextProps.slug))
> > +    }
> > +
> > +    // content has loaded
> > +    if (nextProps.description) {
> > +      this.hideTimeoutId = setTimeout(() => this.hide(), 3000)
>
> does this mean that if you mouse over a link but don't move the mouse onto
> the popover, it will disappear?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Hylozoic/hylo-redux/pull/63/files/d465fd1fd1ffc2a58ffde8eed20f2bdaefdf77c1#r67071801>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA2Y9N58cCUbZ8VM2ms6ZnOTSsuIoTJUks5qLy8ggaJpZM4I1ybD>
> .
>



-- 
Robbie Carlton

719.358.3482
<robbie@authenticmanprogram.com>